tests/chainerx_tests/unit_tests/routines_tests/test_linalg.py

```diff
@@ -93,3 +93,117 @@ def test_dot_invalid(is_module, xp, device, a_shape, b_shape, dtype):
         return xp.dot(a, b)
     else:
         return a.dot(b)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({
+        'shape': [(1, 1), (3, 3), (6, 6)],
+        'in_dtypes': ['float32', 'float64'],
+    })
+))
+class TestSolve(op_utils.NumpyOpTest):
+
+    dodge_nondifferentiable = True
+
+    def setup(self):
+        device = chainerx.get_default_device()
+        if (device.name == 'native:0'
+                and not chainerx.linalg._is_lapack_available()):
+            pytest.skip('LAPACK is not linked to ChainerX')
+
+    def generate_inputs(self):
+        a = numpy.random.random(self.shape).astype(self.in_dtypes)
+        b = numpy.random.random(self.shape[0]).astype(self.in_dtypes)
```

Comment:
The current implementation allows `b` to be higher-dimensional? Should we test them too?

Reply:
Implementation was buggy for that case. It's fixed now and tests are added.